src/mesh/mesh.cpp

```diff
@@ -1050,8 +1053,10 @@ void Mesh::Initialize(bool init_problem, ParameterInput *pin, ApplicationInput *
     // Create send/recv MPI_Requests for all BoundaryData objects
     for (int i = 0; i < nmb; ++i) {
       auto &pmb = block_list[i];
+      // TODO(mpi people) do we still need the pbval part? Discuss also in the context of
+      // other than cellvariables, see comment above on communicators.
```

Comment:
Is there any non-"in one" communication we still do?

Reply:
If I understood your comment above correctly then `RestrictBoundaries` may be the last place.
However, as far as I can tell the following line can independently be removed because the `vbar` ctor already enlists the `vbars` in the `pbval` list.

Reply:
That seems reasonable.